gorm/fieldmask.go

```diff
@@ -36,6 +37,13 @@ func MergeWithMask(source, dest interface{}, mask *fieldmask.FieldMask) error {
 				dstVal = dstVal.Elem()
 				srcVal = srcVal.Elem()
 			}
+			// For safety, skip paths that will cause a panic to call FieldByName on
+			if dstVal.Kind() == reflect.Invalid || dstVal.Kind() == reflect.Array ||
```

Comment:
did you get them all? Too bad there's no `CanFieldByName` like there is `CanAddr`, etc.

Reply:
The [reflect.FieldByName](https://golang.org/pkg/reflect/#Value.FieldByName) documentation leads me to believe that this function will only work on `struct` kinds. Would this be a cleaner test?